### PR TITLE
Add a basic support for validation

### PR DESCRIPTION
**Pull request by gcoutable**
### Type of this PR 

- [x] New feature or improvement

### What does this PR do?

This PR brings Tabs for both right and left panels.
- The right panel contains the explorer in its tab.
- The left panel contains the properties view and the validation view in their respective tabs.
- Also updates the properties view initialization to make it works when the properties view is becoming displayed while there is already a selection, but with a small side effect
- The validation view subscribes to validation refresh events in order to update its content.
- A validation refresh event is sent when any semantic change is performed.

### Screenshot/screencast of this PR

...
 

### Potential side effects

The properties view was not meant to be hidden in the workbench while there is a selection. Since it can be hidden, the properties view blink rapidly from the 'empty' state to the 'ready' state.

https://user-images.githubusercontent.com/7371042/123276550-8eaa2480-d505-11eb-8d2e-113133079c10.mp4

### CODE REVIEW

File: frontend/src/leftSite/leftSite.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
```
[the remark this answers is not in the document]
Comment (sbegaudeau) [reply frontend/src/leftSite/leftSite.tsx:1]: On top of that, could you just move both the left site and right site components and types to the workbench folder please?

File: frontend/src/properties/PropertiesWebSocketContainer.tsx [hunk 2]
```diff
@@ -78,11 +78,7 @@ export const PropertiesWebSocketContainer = ({
   const [{ value, context }, dispatch] = useMachine<
     PropertiesWebSocketContainerContext,
     PropertiesWebSocketContainerEvent
-  >(propertiesWebSocketContainerMachine, {
-    context: {
-      currentSelection: selection,
-    },
-  });
+  >(propertiesWebSocketContainerMachine);
```
Comment (sbegaudeau) [inline frontend/src/properties/PropertiesWebSocketContainer.tsx:81]: I don't mind having this change temporarily but as soon as you're done with the first version of the validation rules, the lifecycle of the properties will needs to be updated to fix this regression.

File: backend/sirius-web-graphql/pom.xml [hunk 3]
```diff
@@ -89,6 +89,11 @@
 			<artifactId>sirius-web-collaborative-trees-api</artifactId>
 			<version>0.3.3</version>
 		</dependency>
+		<dependency>
+			<groupId>org.eclipse.sirius.web</groupId>
+			<artifactId>sirius-web-collaborative-validation-api</artifactId>
+			<version>0.3.3</version>
+		</dependency>
```
Comment (sbegaudeau) [inline backend/sirius-web-graphql/pom.xml:96]: This project depends on sirius-web-graphql-schema and this dependency has been added there already

File: backend/sirius-web-spring-collaborative-validation/pom.xml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+		<dependency>
+			<groupId>org.eclipse.sirius.web</groupId>
+			<artifactId>sirius-web-spring-collaborative</artifactId>
+			<version>0.3.3</version>
+		</dependency>
+		<dependency>
+    		<groupId>org.eclipse.sirius.web</groupId>
+    		<artifactId>sirius-web-tests</artifactId>
+    		<version>0.3.3</version>
+    		<scope>test</scope>
```
Comment (sbegaudeau) [inline backend/sirius-web-spring-collaborative-validation/pom.xml:65]: Formatting issue

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
I still have a couple things to check for this PR but here are my main comments

File: frontend/src/leftSite/leftSite.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gcoutable) [reply frontend/src/leftSite/leftSite.tsx:1]: Done

File: frontend/src/leftSite/leftSite.types.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply frontend/src/leftSite/leftSite.types.ts:1]: Done

File: frontend/src/rightSite/rightSite.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply frontend/src/rightSite/rightSite.tsx:1]: Done

File: frontend/src/rightSite/rightSite.types.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+/*******************************************************************************
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply frontend/src/rightSite/rightSite.types.ts:1]: Done

File: frontend/src/properties/PropertiesWebSocketContainer.tsx (hunk 2 above)
Comment (gcoutable) [reply frontend/src/properties/PropertiesWebSocketContainer.tsx:81]: Issue https://github.com/eclipse-sirius/sirius-components/issues/602 created

File: backend/sirius-web-graphql/pom.xml (hunk 3 above)
Comment (gcoutable) [reply backend/sirius-web-graphql/pom.xml:96]: That means the three dependencies above should be removed too. Not necessary in this PR.

File: backend/sirius-web-spring-collaborative-validation/pom.xml (hunk 4 above)
Comment (gcoutable) [reply backend/sirius-web-spring-collaborative-validation/pom.xml:65]: Done

File: backend/sirius-web-graphql/pom.xml (hunk 3 above)
Comment (gcoutable) [reply backend/sirius-web-graphql/pom.xml:96]: Done

File: frontend/src/index.ts [hunk 8]
```diff
@@ -97,7 +97,11 @@ export * from './tree/TreeItemDocumentContextMenu.types';
 export * from './tree/TreeItemObjectContextMenu';
 export * from './tree/TreeItemObjectContextMenu.types';
 export * from './views/FormContainer';
+export * from './workbench/LeftSite';
+export * from './workbench/LeftSite.types';
 export * from './workbench/RepresentationContext';
 export * from './workbench/RepresentationNavigation';
+export * from './workbench/RightSite';
+export * from './workbench/RightSite.types';
```
Comment (sbegaudeau) [inline frontend/src/index.ts]: No need to export things that are not used outside of Sirius Components

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**